contracts/libraries/VaultLifecycle.sol

```diff
@@ -436,24 +436,28 @@ library VaultLifecycle {
 
     /**
      * @notice Gets the next options expiry timestamp
+     * @param currentExpiry is the expiry timestamp of the current option
+     * Reference: https://codereview.stackexchange.com/a/33532
+     * Examples:
+     * getNextFriday(week 1 thursday) -> week 1 friday
+     * getNextFriday(week 1 friday) -> week 2 friday
+     * getNextFriday(week 1 saturday) -> week 2 friday
      */
     function getNextFriday(uint256 currentExpiry)
         internal
         pure
         returns (uint256)
     {
-        uint256 nextWeek = currentExpiry + 86400 * 7;
-        uint256 dayOfWeek = ((nextWeek / 86400) + 4) % 7;
+        // dayOfWeek = 1 (monday) - 7 (sunday)
+        uint256 dayOfWeek = ((currentExpiry / 86400) + 4) % 7;
+        uint256 nextFriday = currentExpiry + ((7 + 5 - dayOfWeek) % 7) * 86400;
+        uint256 friday8am =
+            nextFriday - (nextFriday % (60 * 60 * 24)) + (8 * 60 * 60);
 
-        uint256 friday;
-        if (dayOfWeek > 5) {
-            friday = nextWeek - 86400 * (dayOfWeek - 5);
-        } else {
-            friday = nextWeek + 86400 * (5 - dayOfWeek);
+        // If the passed currentExpiry is day=Friday hour>8am, we simply increment it by a week to next Friday
+        if (currentExpiry >= friday8am) {
```

Comment:
this is only if currentExpiry is past friday 8am but before that week's sunday?

Reply:
Yep, this scenario only happens on Friday